core/state_processor.go

```diff
@@ -148,7 +148,7 @@ func ApplyTransactionWithEVM(msg *Message, config *params.ChainConfig, gp *GasPo
 
 	// Create a new context to be used in the EVM environment.
 	txContext := NewEVMTxContext(msg)
-	evm.Reset(txContext, tracingStateDB)
+	evm.SetTxContext(txContext)
```

Comment:
So, I think this change throws away the `tracingStateDB`. You need to do something like `evm.SetStateDB(tracingStateDB)` in order to make it use the layered one and not the one it was constructed with. 

...Ah, wait, you are already setting it earlier. So the only thing you are using the `tracingStateDB` for _here_, is to invoke `Finalize` on it ? If so, I think you should replace that with `evm.StateDB.Finalise()`, and drop the notion of `tracingStateDB` in this method

Reply:
I pushed a fix